cppcache/src/ThinClientPoolDM.cpp

```diff
@@ -474,20 +484,19 @@ void ThinClientPoolDM::cleanStaleConnections(std::atomic<bool>& isRunning) {
                 nextIdle < _nextIdle) {
               _nextIdle = nextIdle;
             }
-            savelist.push_back(conn);
+            put(conn, false);
           }
         }
       }
       replaceCount--;
+      count++;
+      if (count % 40 == 0) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
This looks like debug code(?!), please remove

Reply:
This is  part of solution for this issue. Main problem in this issue is that this thread takes over processor for long period (if we have many connections), and due to that we have latency in request processing. So idea is every 40 cycles to give chance to other threads(this is what we tested). Maybe we can try with sleep 1ms every 10 cycles, or use of yield.

Reply:
a couple things:  has a test been created to drive out this issue so it can be seen to be fixed; second, it would be nice to find an alternative to sleeping the thread